scripts/synapse_sytest.sh

```diff
@@ -121,7 +121,7 @@ if [ -n "$OFFLINE" ]; then
 else
     # We've already created the virtualenv, but lets double check we have all
     # deps.
-    /venv/bin/pip install -q --upgrade --no-cache-dir /src
+    /venv/bin/pip install -q --upgrade --no-cache-dir /src[redis]
```

Comment:
Do we want to be testing a setup with redis specified in addition?

Reply:
Basically the situation is that this happens before we call sytest and so don't have access to the config options. We could:

1. Just always install redis deps, the downside is that sytest won't catch if we inadvertently require redis deps when redis isn't configured.
2. Add an env var to configure whether we install redis deps, downside is that it adds more options.

Personally, I'm not sure sytest is the place to test whether optional deps really are optional, especially if we end up wanting to add tests for optional features in sytest we'd need to keep adding more sytest runs (though arguably we shouldn't be testing optional features in sytest).

Reply:
That makes sense. Thanks! 👍